### ISSUE TITLE
[icetray] I3TestModuleFactory throws an error when given more than one test case class (Trac #2189)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#2181]
The `I3TestModuleFactory` throws an error when given more than one test case class:

ERROR (I3Module): I3TestModule_0000: Exception thrown (I3Module.cxx:123 in void I3Module::Do(void (I3Module::*)()))
Traceback (most recent call last):
  File "lilliput/resources/test/test_basic_seed_service.py", line 200, in <module>
    tray.Execute(5)
  File "/home/kai/IceCube/combo/build_debug/lib/I3Tray.py", line 254, in Execute
    super(I3Tray, self).Execute(args[0])
  File "/home/kai/IceCube/combo/build_debug/lib/icecube/icetray/test_module.py", line 67, in RunTests
    result = self.runner.run(test, result)
TypeError: run() takes exactly 2 arguments (3 given)
Exited Icetray Environment.

Migrated from https://code.icecube.wisc.edu/ticket/2189
```json
{
    "status": "closed", 
    "changetime": "2019-02-13T14:15:23", 
    "description": "The `I3TestModuleFactory` throws an error when given more than one test case class:\n\nERROR (I3Module): I3TestModule_0000: Exception thrown (I3Module.cxx:123 in void I3Module::Do(void (I3Module::*)()))\nTraceback (most recent call last):\n  File \"lilliput/resources/test/test_basic_seed_service.py\", line 200, in <module>\n    tray.Execute(5)\n  File \"/home/kai/IceCube/combo/build_debug/lib/I3Tray.py\", line 254, in Execute\n    super(I3Tray, self).Execute(args[0])\n  File \"/home/kai/IceCube/combo/build_debug/lib/icecube/icetray/test_module.py\", line 67, in RunTests\n    result = self.runner.run(test, result)\nTypeError: run() takes exactly 2 arguments (3 given)\nExited Icetray Environment.", 
    "reporter": "kkrings", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1550067323910946", 
    "component": "combo reconstruction", 
    "summary": "[icetray] I3TestModuleFactory throws an error when given more than one test case class", 
    "priority": "normal", 
    "keywords": "", 
    "time": "2018-09-18T10:12:06", 
    "milestone": "", 
    "owner": "kkrings", 
    "type": "defect"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20180918T10:12:29: 
* kkrings commented:

> Proposed patch


* kkrings changed attachment from "" to "test_module.patch"

**Comment by icecube-trac** [comment]
Trac update at 20180922T09:53:20: 
* olivas changed milestone from "" to "Winter Solstice 2018"
* olivas changed status from "new" to "assigned"
* olivas changed owner from "" to "olivas"

**Comment by icecube-trac** [comment]
Trac update at 20181127T15:00:22: olivas changed owner from "olivas" to "kkrings"

**Comment by icecube-trac** [comment]
Trac update at 20181127T15:06:15: kkrings changed status from "assigned" to "accepted"

**Comment by icecube-trac** [comment]
Trac update at 20181205T11:28:25: 
* kkrings changed resolution from "" to "fixed"
* kkrings commented:

> In changeset 3114:
> 
> ```text
> #!CommitTicketReference repository="IceTray" revision="3114"
> Put all test cases into a TestSuite; fixes #2181.
> ```


* kkrings changed status from "accepted" to "closed"

**Comment by icecube-trac** [comment]
Trac update at 20190111T21:05:37: nwhitehorn commented:

> In changeset 168865:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="168865"
> Put all test cases into a TestSuite; fixes #2181.
> ```

**Comment by icecube-trac** [comment]
Trac update at 20190213T14:15:23: 
* olivas changed milestone from "Winter Solstice 2018" to ""
* olivas commented:

> Milestone Winter Solstice 2018 deleted
